pyarrow 21.0.0 breaks audb.stream()

When running the tests under Python 3.13 using `pyarrow==21.0.0` the tests for streaming with a batch size of 0 break with:

```
        if batch_size <= 0:
>           raise ValueError("batch_size must be greater than zero")
E           ValueError: batch_size must be greater than zero

.venv/lib/python3.13/site-packages/pyarrow/parquet/core.py:583: ValueError
```

--- a/audb/core/stream.py
+++ b/audb/core/stream.py
@@ -2,6 +2,7 @@ from __future__ import annotations
 
 import abc
 from collections.abc import Iterable
+from collections.abc import Iterator
 from collections.abc import Sequence
 import os
 
@@ -397,13 +398,12 @@ class DatabaseIteratorCsv(DatabaseIterator):
 
 
 class DatabaseIteratorParquet(DatabaseIterator):
-    def _initialize_stream(self) -> pa.RecordBatch:
+    def _initialize_stream(self) -> Iterator[pa.RecordBatch]:
         if self._samples == 0:
             # pyarrow >=21.0.0 does not support batch_size=0
-            return pa.RecordBatch.from_arrays([])
-        else:
-            file = os.path.join(self.root, f"db.{self._table}.parquet")
-            return parquet.ParquetFile(file).iter_batches(batch_size=self._samples)
+            return []
+        file = os.path.join(self.root, f"db.{self._table}.parquet")
+        return parquet.ParquetFile(file).iter_batches(batch_size=self._samples)
 
     def _postprocess_batch(self, batch: pa.RecordBatch) -> pd.DataFrame:
         df = batch.to_pandas(

--- a/audb/core/stream.py
+++ b/audb/core/stream.py
@@ -398,8 +398,12 @@ class DatabaseIteratorCsv(DatabaseIterator):
 
 class DatabaseIteratorParquet(DatabaseIterator):
     def _initialize_stream(self) -> pa.RecordBatch:
-        file = os.path.join(self.root, f"db.{self._table}.parquet")
-        return parquet.ParquetFile(file).iter_batches(batch_size=self._samples)
+        if self._samples == 0:
+            # pyarrow >=21.0.0 does not support batch_size=0
+            return pa.RecordBatch.from_arrays([])
+        else:
+            file = os.path.join(self.root, f"db.{self._table}.parquet")
+            return parquet.ParquetFile(file).iter_batches(batch_size=self._samples)
 
     def _postprocess_batch(self, batch: pa.RecordBatch) -> pd.DataFrame:
         df = batch.to_pandas(

